proto/proto/block_producer.proto

```diff
@@ -0,0 +1,11 @@
+// Specification of the user facing gRPC API.
+syntax = "proto3";
+package block_producer;
+
+import "requests.proto";
+import "responses.proto";
+
+service Api {
```

Comment:
I'd use a more descriptive name, such as `BlockProducer`

Reply:
I think this may be fine because this is under `block_producer` package (and this is more consistent with how we do it for the store and RPC components). Though, I wonder if we should put all API definitions under `api` package (e.g., `api.proto`). Let's create an issue to evaluate pros and cons of this approach.

Reply:
Ah, just realized that. I agree that this is a good convention with "well named" packages, such that in Rust we always use e.g. `block_producer_client::ApiClient`. The one instance I find unintuitive is actually when we [use the RPC's `ApiClient`](https://github.com/0xPolygonMiden/miden-node/blob/ec80583b4b77406529accb39a2b0edf39f8da2f8/rpc/src/main.rs#L30); i.e. `api_client::ApiClient` is not very descriptive IMO.

Reply:
That is generated code, [ref](https://github.com/0xPolygonMiden/miden-node/blob/ec80583b4b77406529accb39a2b0edf39f8da2f8/proto/src/generated/rpc.rs#L1-L21). We could:

- Do what bobbing is suggesting, and swap things around.
    - move rpc/block_producer/store to a single api.proto. Then the name above would be `rpc_client::RpcClient`.
- re-export the generated `api_client::ApiClient` as `rpc_client::Client` or something like that